server/webhooks/failing_policies.go

```diff
@@ -0,0 +1,186 @@
+package webhooks
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"net/url"
+	"path"
+	"sort"
+	"strconv"
+	"time"
+
+	"github.com/fleetdm/fleet/v4/server"
+	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	kitlog "github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+)
+
+// TriggerGlobalFailingPoliciesWebhook performs the webhook requests for failing policies.
+func TriggerGlobalFailingPoliciesWebhook(
+	ctx context.Context,
+	ds fleet.Datastore,
+	logger kitlog.Logger,
+	appConfig *fleet.AppConfig,
+	failingPoliciesSet fleet.FailingPolicySet,
+	now time.Time,
+) error {
+	if !appConfig.WebhookSettings.FailingPoliciesWebhook.Enable {
+		return nil
+	}
+
+	level.Debug(logger).Log("enabled", "true")
+
+	serverURL, err := url.Parse(appConfig.ServerSettings.ServerURL)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "invalid server url")
+	}
+	globalPoliciesURL := appConfig.WebhookSettings.FailingPoliciesWebhook.DestinationURL
+	if globalPoliciesURL == "" {
+		level.Info(logger).Log("msg", "empty global destination_url")
+		return nil
+	}
+	policies, err := filterPolicies(ctx, ds,
+		appConfig.WebhookSettings.FailingPoliciesWebhook.PolicyIDs,
+		failingPoliciesSet,
+		logger,
+	)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "filtering policies")
+	}
+	for _, policy := range policies {
+		if err := sendBatchedPOSTs(ctx, policy, failingPoliciesSet, postData{
+			serverURL:  serverURL,
+			now:        now,
+			webhookURL: globalPoliciesURL,
+		}, appConfig.WebhookSettings.FailingPoliciesWebhook.HostBatchSize, logger); err != nil {
+			return ctxerr.Wrapf(ctx, err, "sending POSTs for policy set %d", policy.ID)
+		}
+	}
+	return nil
+}
+
+type postData struct {
+	serverURL  *url.URL
+	now        time.Time
+	webhookURL string
+}
+
+func sendBatchedPOSTs(
```

Comment:
This reads more generic than it is. I was expecting something that would batch any request, not just failing policies ones. So maybe rename it to `sendFailingPoliciesBatchedPosts` or something like that?